### ISSUE TITLE
Cannot Apply Free Shipping Based on Cart Contents Total/Subtotal When Coupons Are Used

### ISSUE THREAD

**Issue by alexp88** [woocommerce/woocommerce#12819]
Greetings,

I have been trying to get WooCommerce to apply free shipping based on cart contents total/subtotal rather than total when coupons are used, however, it currently enables free shipping only when the order total/cart total reaches the minimum threshold with the discount in place.

I have the following Shipping Methods enabled (created in exactly the same order as below):
1. Free UK Standard Delivery (over 30GBP)
2. Free UK Next Working Day Delivery (over 60GBP)
3. UK Standard Delivery
4. UK Next Working Day Delivery
5. UK Saturday Delivery

Methods 1 & 3 and 2 & 4 are essentially the same, with the only difference being that 1 and 2 are available if a minimum order amount is reached (that is, 30 and respectively 60GBP). Currently, WooCommerce doesn't offer an option to turn off other shipping options when a free shipping method is enabled (i.e. turn off option no. 3 if no. 1 is enabled), or make any of the available shipping methods free if a certain threshold is reached, so I had to come up with a solution. Here's the code I added to my functions.php file:

```
add_filter( 'woocommerce_package_rates', 'bbloomer_unset_shipping_when_free_is_available_in_zone', 10, 2 );
  
function bbloomer_unset_shipping_when_free_is_available_in_zone( $rates, $package ) {
     
    // Only unset rates if free_shipping is available
    if ( isset( $rates['free_shipping:7'] ) ) {
    unset( $rates['flat_rate:9'] );
}   
     
return $rates;
 
}

add_filter( 'woocommerce_package_rates', 'bbloomer_unset_shipping_when_free_is_available_in_zone2', 10, 2 );
  
function bbloomer_unset_shipping_when_free_is_available_in_zone2( $rates, $package ) {
     
    // Only unset rates if free_shipping is available
    if ( isset( $rates['free_shipping:8'] ) ) {
    unset( $rates['flat_rate:9'] );
    unset( $rates['flat_rate:10'] );
    unset( $rates['free_shipping:7'] );
}   
     
return $rates;
 
}
```

The first filter disables UK Standard Delivery when Free UK Standard Delivery is available, while the second one disables UK Standard Delivery, UK Next Working Day Delivery, and Free UK Standard Delivery when Free UK Next Working Day Delivery is available. WooCommerce displays the first shipping method as default, so by ordering them as above, I made sure that the Free option is always set as default when the minimum order amount condition is met. The customer can only "upgrade" (to UK Next Working Day Delivery and UK Saturday Delivery if Free UK Standard Delivery is available, and to UK Saturday Delivery if Free UK Next Working Day Delivery is available).

Long story short, the above is working perfectly and does what it should, however, the threshold for each of the two free shipping methods is the order total after the coupon code is applied. So even if the cart contents total > threshold, free shipping still won't be available unless the order total (discount included) is also > threshold.

Is there a workaround to prevent this and apply free shipping based on cart contents total rather than order total?

FYI, I'm running WooCommerce 2.6.12.

**Comment by mikejolley** [comment]
I suggest you use this filter instead of changing packages: https://github.com/woocommerce/woocommerce/blob/master/includes/shipping/free-shipping/class-wc-shipping-free-shipping.php#L194

You can then run logic to check the price and disable methods if they do or do not meet the criteria you set for the cart.

This is not a bug report so closing the issue.
